e2e/lookout_ingester_test/lookout_ingester_test.go

```diff
@@ -456,3 +472,27 @@ func generateNotificationConnectionEventHandler(channelName string) func(pq.List
 		}
 	}
 }
+
+var (
+	createShadowTableSql = "CREATE TABLE %s_shadow (LIKE %s INCLUDING ALL, update_num SERIAL);"
+	alterShadowTableSql  = "ALTER TABLE %s_shadow DROP CONSTRAINT %s_shadow_pkey, ADD PRIMARY KEY (job_id, update_num);"
+)
+
+func createShadowTables(db *sql.DB) {
```

Comment:
you might be able to remove some of this boilerplate with https://www.postgresql.org/docs/14/sql-notify.html. I've actually never used this so I'm not 100% sure whether it will actually make your life easier though.

Reply:
I think I prefer this way instead of using notify. This has the added benefit of creating a durable log of what changed during test execution. Notify is more ephemeral.